airflow-core/src/airflow/utils/static_checker.py

```diff
@@ -0,0 +1,468 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import ast
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+
+
+@dataclass
+class RuntimeVaryingValueWarning:
+    """Warning information for runtime-varying value detection."""
+
+    line: int
+    col: int
+    code: str
+    message: str
+
+
+class WarningContext(str, Enum):
+    """Context types for warnings."""
+
+    TASK_CONSTRUCTOR = "TASK constructor"
+    DAG_CONSTRUCTOR = "DAG constructor"
```

Comment:
```suggestion
    TASK_CONSTRUCTOR = "Task constructor"
    DAG_CONSTRUCTOR = "Dag constructor"
```

Is there a reason we use upper case?